[Bug]: Build fails on Windows because of wrong file separators

### Is there an existing issue for this?

- [X] I have searched the existing issues

### In which area is the issue?

General

### In a specific sample?

_No response_

### What happened?

The build fails because samples are missing, and Hilt cannot inject.

My solution is to replace `/` in [settings.build.kts](https://github.com/android/platform-samples/blob/54e8d71610a7d85aabb8d28495647375669661a2/settings.gradle.kts#L38) with `File.separator`. And then everything works as expected. I don't know if it's the correct solution. If so, I'm also happy to open a PR.

### Relevant logcat output

```shell
platform-samples\app\build\generated\hilt\component_sources\debug\com\example\platform\app\MainApp_HiltComponents.java:133: error: [Dagger/MissingBinding] java.util.Set<com.google.android.catalog.framework.base.CatalogSample> cannot be provided without an @Provides-annotated method.
  public abstract static class SingletonC implements MainApp_GeneratedInjector,
                         ^
      java.util.Set<com.google.android.catalog.framework.base.CatalogSample> is injected at
          com.google.android.catalog.framework.ui.CatalogActivity.catalogSamples
      com.example.platform.app.MainActivity is injected at
          com.example.platform.app.MainActivity_GeneratedInjector.injectMainActivity(com.example.platform.app.MainActivity) [com.example.platform.app.MainApp_HiltComponents.SingletonC �� com.example.platform.app.MainApp_HiltComponents.ActivityRetainedC �� com.example.platform.app.MainApp_HiltComponents.ActivityC]
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -35,14 +35,15 @@ rootProject.name = "Platform Samples"
 include(":app")
 
 // Dynamically include samples under /app-catalog/samples/ folder
-val samples = buildList<String> {
+val samples = buildList {
+    val separator = File.separator
     // Find all build.gradle files under samples folder
     settingsDir.walk()
         .filter { it.name == "build.gradle" || it.name == "build.gradle.kts" }
-        .filter { it.path.contains("/samples/") }
+        .filter { it.path.contains("${separator}samples${separator}") }
         .map { it.parent.substring(rootDir.path.length) }
         .forEach {
-            add(it.replace("/", ":"))
+            add(it.replace(separator, ":"))
         }
 }
 

